phoenix-pherf/src/it/java/org/apache/phoenix/pherf/PherfMainIT.java

```diff
@@ -43,4 +64,83 @@ public void testPherfMain() throws Exception {
             future.get();
         }
     }
+
+    @Test
+    public void testQueryTimeout() throws Exception {
+        // Timeout of 0 ms means every query execution should time out
+        String[] args = {"-q", "-l",
+                "-drop", "all",
+                "-schemaFile", ".*timeout_test_schema.sql",
+                "-scenarioFile", ".*timeout_test_scenario.xml" };
+        Pherf p = new Pherf(args);
+        p.run();
+
+        CSVFileResultHandler rh = new CSVFileResultHandler();
+        rh.setResultFileDetails(ResultFileDetails.CSV_DETAILED_PERFORMANCE);
+        rh.setResultFileName("COMBINED");
+        List<Result> resultList = rh.read();
+        for (Result r : resultList) {
+            HashMap<String, String> resultsMap = mapResults(r);
+            if (resultsMap.get("QUERY_ID").equals("q1")) {
+                assertEquals(resultsMap.get("TIMED_OUT"), "true");
+            }
+        }
+    }
+
+    @Test
+    public void testLargeQueryTimeout() throws Exception {
+        // Timeout of max_long ms means every query execution should time out
+        String[] args = {"-q", "-l",
+                "-drop", "all",
+                "-schemaFile", ".*timeout_test_schema.sql",
+                "-scenarioFile", ".*timeout_test_scenario.xml" };
+        Pherf p = new Pherf(args);
+        p.run();
+
+        CSVFileResultHandler rh = new CSVFileResultHandler();
+        rh.setResultFileDetails(ResultFileDetails.CSV_DETAILED_PERFORMANCE);
+        rh.setResultFileName("COMBINED");
+        List<Result> resultList = rh.read();
+        for (Result r : resultList) {
+            HashMap<String, String> resultsMap = mapResults(r);
+            if (resultsMap.get("QUERY_ID").equals("q2")) {
+                assertEquals(resultsMap.get("TIMED_OUT"), "false");
+            }
+        }
+    }
+
+    @Test
+    public void testNoQueryTimeout() throws Exception {
+        // Missing timeout attribute means every query execution should time out
+        String[] args = {"-q", "-l",
+                "-drop", "all",
+                "-schemaFile", ".*timeout_test_schema.sql",
+                "-scenarioFile", ".*timeout_test_scenario.xml" };
+        Pherf p = new Pherf(args);
+        p.run();
+
+        CSVFileResultHandler rh = new CSVFileResultHandler();
+        rh.setResultFileDetails(ResultFileDetails.CSV_DETAILED_PERFORMANCE);
+        rh.setResultFileName("COMBINED");
+        List<Result> resultList = rh.read();
+        for (Result r : resultList) {
+            HashMap<String, String> resultsMap = mapResults(r);
+            if (resultsMap.get("QUERY_ID").equals("q3")) {
+                assertEquals(resultsMap.get("TIMED_OUT"), "false");
+            }
+        }
+    }
+
+    @Test
+    public void testDiffTimeout() throws Exception {
```

Comment:
I'd like to write an IT for MultiThreadedDiffer, but it's not set up to write any output files, which makes it difficult to verify the results. Looking into adding output functionality to it, but the structure of Differ is quite different than Runner

Reply:
Like discussed online, let's avoid changes to the Differ since it probably isn't too useful to time queries in that mode anyways. Please correct me if you think that's not the case @virajjasani @mihir6692